### PR TITLE
Rewrite the graphical diagram using MermaidJS

### PR DESCRIPTION
**Pull request by sdball**
GitHub allows writing MermaidJS diagrams in READMEs so this may be a nicer way to present the process diagram. MermaidJS has an ER diagram type as well which is arguably a more accurate way to describe the model here, but I personally find the flowchart type to be more generally readable.

image

### CODE REVIEW

**Comment by ryanwinchester** [comment]
This is cool, and a nice feature for github.

However, NGL, I like the text graphics since it visually works _everywhere_.

**Comment by sdball** [comment]
Yeah that’s fair. The mermaid diagram is _readable_ but not visual from text editors.

File: README.md [hunk 1]
```diff
@@ -115,17 +115,21 @@ autonomously, improving scalability and crash resiliency.
...
-                                    / ....n.... \
-                                 Handler (gen_server)
+```mermaid
+graph TD;
+  Server(Server: supervisor, rest_for_one)-->Listener;
+  Server-->AcceptorPoolSupervisor(AcceptorPoolSupervisor: dynamic supervisor);
+  AcceptorPoolSupervisor--1-->AcceptorSupervisor(AcceptorSupervisor: supervisor, rest_for_one)
+  AcceptorPoolSupervisor--2-->AcceptorSupervisor
+  AcceptorPoolSupervisor--3-->AcceptorSupervisor
+  AcceptorPoolSupervisor--n-->AcceptorSupervisor
```
Comment (mtrudel) [inline README.md:125]:
```suggestion
  AcceptorPoolSupervisor--1...n-->AcceptorSupervisor(AcceptorSupervisor: supervisor, rest_for_one)
```

File: README.md [hunk 2]
```diff
@@ -115,17 +115,21 @@ autonomously, improving scalability and crash resiliency.
...
+  AcceptorSupervisor-->Acceptor(Acceptor: task)
+  AcceptorSupervisor-->DynamicSupervisor
+  DynamicSupervisor--1-->Handler(Handler: gen_server)
+  DynamicSupervisor--2-->Handler(Handler: gen_server)
+  DynamicSupervisor--3-->Handler(Handler: gen_server)
+  DynamicSupervisor--n-->Handler(Handler: gen_server)
```
Comment (mtrudel) [inline README.md:131]:
```suggestion
  DynamicSupervisor--1...n-->Handler(Handler: gen_server)
```

**Review by mtrudel — CHANGES_REQUESTED** [review]
I like this! Small tidy suggestion pending

**Comment by sdball** [comment]
I like that change! For reference here it results in the following

image

**Comment by ryanwinchester** [comment]
> I like that change! For reference here it results in the following

Or see it at https://github.com/mtrudel/thousand_island/blob/2937161171efab47c5cc5c130be9b72da4f31251/README.md

**Comment by sdball** [comment]
For sure you can render the file, but for visual feedback it's kinda nice to have it inline.

**Comment by mtrudel** [comment]
Thanks for the PR!

**Pull request merged**